joplin/js/editor.js

```diff
@@ -203,46 +169,43 @@ $(function() {
     replaceLanguageLabels();
 
     // TODO: refactor into a function, evaluate performance
-    // have better variable namepsace seperation
     let labelList = state.languageLabels;
-    for (let label in labelList) {
-      if (labelList[label].querySelector) {
-        let languageTag = labelList[label].querySelector('span').innerText;
-        // these seem to be nested twice, from the title to the containing element
-        // TODO: come up with a more elegant and maintable way to check what elements ought to be hidden
+    for (let label of labelList) {
+      if (label.querySelector) {
+        let languageTag = label.querySelector('span').innerText;
+        // TODO: come up with a more elegant and maintainable way to check what elements ought to be hidden
         if (
-          labelList[label].parentElement.parentElement.parentElement.classList
-            .value !== 'struct-block'
+          label.parentElement.parentElement.parentElement.classList
+            .value !== 'rah-static rah-static--height-auto c-sf-block__content'
```

Comment:
so is the TL;DR here that we found some more specific classes to select by? And/or does react-streamfield have different class names for the html here?

Mostly just curious, but also just wondering what sort of changes to watch out for that would potentially break this (if the html template changes, if we end up nesting streamfields in a different structure?

I wonder if we could add a code comment here...before it was more obvious that this was some logic specifically to handle labels that weren't in a struct block, but its less clear what this is for now

Reply:
very good points. What happened was the react-streamfield didnt have the class name as "struct-block" anymore, and when I first started working on this, I just selected the entire class name that was around the struct-block. 
I will update with a code comment explaining what this is for. I am thinking that even if we nest the streamfield in a different structure, the streamfied will still have the same classes. This could break if we somehow slipped another element between the outer one and the fields.